### DISCUSSION TITLE
What would the best approach be for grabbing images from Mechanical?

### DISCUSSION THREAD

**Discussion by tlewis3348** [ansys/pymechanical#820] [Q&A] [answered]
I'm trying to grab images from Mechanical for use in reports. Since I'm just getting started, I'm trying to figure out how to approach this. Namely, would I be better off starting Mechanical in Remote or Embedded mode?

Also, is it possible to permit the user to interact with the instance of Mechanical that is currently being used by Python? I mean, ideally, I would like to be able to have Python access an instance of Mechanical that the user currently has open, but if that isn't possible, I would like to allow the user to interact with the model they are accessing with the tool I'm developing.

**Comment by dipinknair** [comment, marked as answer]
Hi @tlewis3348 , Both Remote and Embedded mode has export image options. It depends on various factors. Are you working with Mechanical locally? do you want to work with Mechanical GUI?  I would suggest going over the examples section of pymechanical docs to see examples both types. With any mode, you can interact with mechanical instance from python. With Embedding you can directly export the image locally, but with Remote it has to be exported to Remote location and then download locally. 
[Remote method example](https://examples.mechanical.docs.pyansys.com/examples/00_basic/example_02_capture_images.html#sphx-glr-examples-00-basic-example-02-capture-images-py)
[Embedding method example](https://embedding.examples.mechanical.docs.pyansys.com/examples/basic/bolt_pretension.html#sphx-glr-examples-basic-bolt-pretension-py)

**Reply by tlewis3348** [reply]
Thanks. That's helpful. However, when I execute the code in the "Embedding" example, I get this error message:

```
import ansys.mechanical.core as mech
app = mech.App(version=242)
app.update_globals(globals())

CRITICAL -  -  logging - handle_exception - Uncaught exception
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'App' object has no attribute 'update_globals'
```

**Reply by dipinknair** [reply]
Can you check whether you are using the latest version of pymechanical ? if now please update ``pip install ansys.mechanical.core --upgrade``. Let me know that solve the issue

**Reply by tlewis3348** [reply]
So, I just installed PyMechanical a couple days ago, but I think something got messed up at some point when I was experimenting with different things. So, I uninstalled and reinstalled. It seems to be working now.

Another question: I'd like to use the embedded method since that appears to be the most straightforward means of doing what I'm trying to do (i.e., it doesn't require the user to archive the model thereby duplicating the space consumed by the model), but I'd like the user to be able to provide the WBPJ file and select the name/letter of the system with the mechanical model they want to access. Since PyWorkbench appears to require the use of the remote method, that doesn't seem like a viable option for me. However, it seems like if I know the correct place to look in the XML of the WBPJ file, I should be able to get a list of the systems in the file to offer to the user as a list to select from. My question therefore is, can you provide any guidance on how to navigate the XML of the WBPJ file to get a list of the systems in the file and the location of their MECHDB files?

P.S. I'm not sure if this is the correct place to be asking these questions. Should I be going to the customer support portal and creating a case there? Alternatively, should I be creating a separate question for each of these instead of replying to the original question?

**Reply by koubaa** [reply]
@tlewis3348 This place is fine for discussions about PyMechanical specifically, but we also take questions for Mechanical and other applications on the ansys developer portal (developer.ansys.com).

PyWorkbench is one way to interact with workbench, you also have the option to run WorkBench by command line, and use the scripting capabilities of WorkBench to scrape the project for information about the Mechanical systems and their respective mechdb files. I'm not familiar enough with WorkBench scripting to attempt to write a snippet, but I expect that it is possible. I don't know much about the WBPJ file format or how stable it is (i.e. if a project saved in 2019R1 and 2023R2 are similar enough that the same code to read it will work?)

**Comment by pmaroneh** [comment]
To add to the inputs already provided, this example might help: https://developer.ansys.com/blog/create-report-saved-mechanical-session-mechdb-or-mechdat

**Reply by tlewis3348** [reply]
The ADR method of report generation is interesting, but it will require a significant lateral change in our workflows. So, while I would personally like to fully switch over to that report format, my guess is that it will be extremely difficult to get any significant percentage of others on board. Therefore, my current plan is to automate the generation of PowerPoint reports.

**Reply by pmaroneh** [reply]
Totally understandable. I'd encourage you to create an issue in the PyADR repo to ask for powerpoint capability.

**Reply by tlewis3348** [reply]
Looks like someone beat me to it, and it also looks like Ansys beat me to doing what I was wanting to do: [https://github.com/ansys/pydynamicreporting/issues/112#issuecomment-2245213668](https://github.com/ansys/pydynamicreporting/issues/112#issuecomment-2245213668). Here's the documentation: [https://nexusdemo.ensight.com/docs/en/html/PPTX.html](https://nexusdemo.ensight.com/docs/en/html/PPTX.html). The feature is in beta at the moment, but it is already far beyond what I've developed, and it has the added advantage of enabling the easy transition to the more feature-rich HTML reports.
